src/shared/dissect-image.c

```diff
@@ -1289,8 +1295,37 @@ static int verity_partition(
 #endif
         } else
                 r = crypt_activate_by_volume_key(cd, name, root_hash, root_hash_size, CRYPT_ACTIVATE_READONLY);
-        if (r < 0)
+        if (r < 0 && r != -EEXIST)
                 return r;
+        if (r < 0) {
```

Comment:
This check (while correct) looks strange. Maybe use `r == -EEXIST` ?

Reply:
Sure, done